src/main/java/gov/usgs/wma/waterdata/JsonDataDao.java

```diff
@@ -59,12 +63,20 @@ public void doGrades(Long jsonDataId) {
 	}
 
 	@Transactional
-	public TimeSeries doHeaderInfo(Long jsonDataId) {
-		return jdbcTemplate.queryForObject(
-				getSql(headerInfo),
-				new TimeSeriesRowMapper(),
-				jsonDataId
-			);
+	public String doHeaderInfo(Long jsonDataId) {
+		try {
+			return jdbcTemplate.queryForObject(
+					getSql(headerInfo),
+					String.class,
+					jsonDataId
+				);
+		} catch (EmptyResultDataAccessException e) {
+			LOG.info("Couldn't find {} - {}", jsonDataId, e.getLocalizedMessage());
+			//Eat the no data exception if the JSON data is not found or
+			//has no time series unique ID. We cannot recover from this and
+			//should not trigger a state machine retry.
+			return null;
```

Comment:
Is this the step that would generate the input that feeds into the map state? So the map would get a `null`?

Reply:
Yes, this should send an empty list to the map state and the map should realize it has no work to do (hopefully). The PreProcessTest.notFoundTest() should mock the behavior in this code.

Reply:
I've tested the map state's behavior with an empty list: it doesn't do anything and reports a success.